cmd/oras/manifest/fetch.go

```diff
@@ -72,46 +79,74 @@ Example - Fetch manifest with prettified json result:
 		},
 	}
 
-	cmd.Flags().BoolVarP(&opts.pretty, "pretty", "", false, "output prettified manifest")
-	cmd.Flags().BoolVarP(&opts.fetchDescriptor, "descriptor", "", false, "fetch a descriptor of the manifest")
 	cmd.Flags().StringSliceVarP(&opts.mediaTypes, "media-type", "", nil, "accepted media types")
+	cmd.Flags().StringVarP(&opts.outputPath, "output", "o", "", "output file path")
 	option.ApplyFlags(&opts, cmd.Flags())
 	return cmd
 }
 
-func fetchManifest(opts fetchOptions) error {
+func fetchManifest(opts fetchOptions) (fetchErr error) {
 	ctx, _ := opts.SetLoggerLevel()
-	targetPlatform, err := opts.Parse()
-	if err != nil {
-		return err
-	}
+
 	repo, err := opts.NewRepository(opts.targetRef, opts.Common)
 	if err != nil {
 		return err
 	}
+
 	if repo.Reference.Reference == "" {
 		return oerrors.NewErrInvalidReference(repo.Reference)
 	}
 	repo.ManifestMediaTypes = opts.mediaTypes
 
-	// Fetch and output
-	var content []byte
-	if opts.fetchDescriptor {
-		content, err = cas.FetchDescriptor(ctx, repo, opts.targetRef, targetPlatform)
-	} else {
-		content, err = cas.FetchManifest(ctx, repo, opts.targetRef, targetPlatform)
+	targetPlatform, err := opts.Parse()
+	if err != nil {
+		return err
 	}
+
+	manifests, err := opts.CachedTarget(repo.Manifests())
 	if err != nil {
 		return err
 	}
-	if opts.pretty {
-		buf := bytes.NewBuffer(nil)
-		if err = json.Indent(buf, content, "", "  "); err != nil {
-			return fmt.Errorf("failed to prettify: %w", err)
+
+	var desc ocispec.Descriptor
+	var content []byte
+	if opts.OutputDescriptor && opts.outputPath == "" {
```

Comment:
I couldn't really follow your idea here.

In the case of fetching manifest content, there still might be a need to output descriptor, eg.  `--output manifest.json --descriptor`. Since `oras.FetchBytes` returns both the content and the descriptor, so if content has already been fetched, we don't need to call `oras.Resolve` again.

Current logic:
```
if opts.OutputDescriptor && opts.outputPath == "" {
	// oras.Resolve   fetch manifest descriptor only
} else {
	// oras.FetchBytes   fetch manifest descriptor content

	if opts.outputPath == "" || opts.outputPath == "-" {
		// output manifest content

		return
	}

	// save manifest content into the local file
}

if opts.OutputDescriptor {
	// output manifest descriptor
}
```
In this way, for `--output manifest.json --descriptor`, there's a redundant `oras.Resolve`.
```
if !opts.OutputDescriptor || opt.outputPath != "" {
	// oras.FetchBytes   fetch manifest descriptor content

	if opts.outputPath == "" || opts.outputPath == "-" {
		// output manifest content

		return
	}

	// save manifest content into the local file
} 

if opts.OutputDescriptor {
	// oras.Resolve   fetch manifest descriptor only

	// output manifest descriptor
}
```

Reply:
> Since oras.FetchBytes returns both the content and the descriptor, so if content has already been fetched, we don't need to call oras.Resolve again

This can be avoided by checking if the desc is a empty struct.